diff --git a/src/configs/starship_root.rs b/src/configs/starship_root.rs
index 6cc4da77..ecf8bcc1 100644
--- a/src/configs/starship_root.rs
+++ b/src/configs/starship_root.rs
@@ -17,6 +17,7 @@ pub struct StarshipRootConfig {
     pub scan_timeout: u64,
     pub command_timeout: u64,
     pub add_newline: bool,
+    pub add_separator: bool,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub palette: Option<String>,
     pub palettes: HashMap<String, Palette>,
@@ -121,6 +122,7 @@ impl Default for StarshipRootConfig {
             scan_timeout: 30,
             command_timeout: 500,
             add_newline: true,
+            add_separator: false,
             palette: None,
             palettes: HashMap::default(),
         }
diff --git a/src/print.rs b/src/print.rs
index 6bb1f1be..a1010c2e 100644
--- a/src/print.rs
+++ b/src/print.rs
@@ -119,6 +119,10 @@ pub fn get_prompt(context: Context) -> String {
         // continuation prompts normally do not include newlines, but they can
         writeln!(buf).unwrap();
     }
+    if config.add_separator && context.width > 0 {
+        let line = std::iter::repeat("═").take(context.width).collect::<String>();
+        write!(buf, "{}\n", line).unwrap();
+    }
     write!(buf, "{}", AnsiStrings(&module_strings)).unwrap();
 
     if context.target == Target::Right {
