diff --git a/src/configs/starship_root.rs b/src/configs/starship_root.rs
index e386b76a..e858b540 100644
--- a/src/configs/starship_root.rs
+++ b/src/configs/starship_root.rs
@@ -12,6 +12,7 @@ pub struct StarshipRootConfig {
     pub scan_timeout: u64,
     pub command_timeout: u64,
     pub add_newline: bool,
+    pub add_separator: bool,
 }
 
 // List of default prompt order
@@ -107,6 +108,7 @@ impl<'a> Default for StarshipRootConfig {
             scan_timeout: 30,
             command_timeout: 500,
             add_newline: true,
+            add_separator: false,
         }
     }
 }
diff --git a/src/print.rs b/src/print.rs
index e624bd5d..71cf47e5 100644
--- a/src/print.rs
+++ b/src/print.rs
@@ -117,6 +117,10 @@ pub fn get_prompt(context: Context) -> String {
         // continuation prompts normally do not include newlines, but they can
         writeln!(buf).unwrap();
     }
+    if config.add_separator && context.width > 0 {
+        let line = std::iter::repeat("═").take(context.width).collect::<String>();
+        write!(buf, "{}\n", line).unwrap();
+    }
     write!(buf, "{}", ANSIStrings(&module_strings)).unwrap();
 
     if context.target == Target::Right {
